aten/src/ATen/native/mps/operations/View.mm

```diff
@@ -68,7 +68,7 @@
                                               dataType:inputType] autorelease];
     if (needsScatter) {
       auto updatesType = getMPSScalarType(src.scalar_type());
-      if (updatesType == MPSDataTypeUInt8 || (updatesType == MPSDataTypeBool && !is_macos_13_or_newer())) {
```

Comment:
I think the point is to cast bool to int8 on macOS 14 (perhaps 13 too?) when dim > 5. On macOS 12, the bug is not triggered since we did the casting. This function [createViewGraph ](https://github.com/pytorch/pytorch/blob/cd1dc5e48477434025e4bdf3827945c5035f6275/aten/src/ATen/native/mps/operations/View.mm#L684) is only called when dim > 5, so no point to check it again (adding it won't fix the bug)